### PR TITLE
Just comments on many files

### PR DESCRIPTION
**Pull request by joaoburatto**
### Summary

Comments on many files to make sure everyone understands whats going on, still a lot to do.
This does not alters code, just simple comments.

### CODE REVIEW

File: Assets/Content/Creatures/Creature.cs [hunk 1]
```diff
@@ -5,10 +5,17 @@
 using UnityEngine;
 using UnityEngine.XR;
 
+// TODO: First and foremost, I should rename this to Entity.
+
+// An entity should be used for anything that ressembles the creature idea
+// it can walk, interact with stuff, be controlled by a player
```
Comment (cosmiccoincidence) [inline Assets/Content/Creatures/Creature.cs:11]: Would this include ghosts too? If so, I'd just put "controlled by player" and remove the walking, interacting, & combat.

File: Assets/Content/Furniture/Machines/Atmospherics/Canister/Canister.cs [hunk 2]
```diff
@@ -9,19 +9,34 @@
 using UnityEngine.Tilemaps;
 
 namespace SS3D.Content.Furniture.Machines.Atmospherics {
+    // This handles the gas canister process, where we can put only Atmos stuff.
+    // This can only store gas (for now), and we have a maximum ammount of gas that can be stored.
+    // You can open the valve to release gas and you can set up the pressure the gas leak
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Machines/Atmospherics/Canister/Canister.cs:14]: "You can open the valve to release gas and you can also set the release pressure"
this is more clear.

File: Assets/Content/Furniture/Machines/Atmospherics/Canister/Canister.cs [hunk 3]
```diff
@@ -80,11 +98,15 @@ public void Initialize()
             return;
         }
 
+	// This handles the gas leaking
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Machines/Atmospherics/Canister/Canister.cs:101]: "This handles the gas releasing"

File: Assets/Content/Furniture/Machines/Atmospherics/Canister/CanisterUI.cs [hunk 4]
```diff
@@ -4,17 +4,26 @@
...
 {
+    // Canister name
     public TMP_Text label;
+    // Current pressure
     public TMP_Text pressure;
+
+    // if it is connected to a pipe or not
     public TMP_Text portStatus;
 
+    // I don't know what this is
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Machines/Atmospherics/Canister/CanisterUI.cs:19]: @henkhooft

File: Assets/Content/Furniture/Machines/Atmospherics/Disposal Bin/DisposalBin.cs [hunk 5]
```diff
@@ -11,16 +11,24 @@
...
+
+	// is the bin dumping trash right now?
         public bool busy = false;
         
         public AudioSource audioSource;
 
         public float range;
 
+	// Interaction that handles the disposal of items
+	// This should be worked once we have disposals pipes goin
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Machines/Atmospherics/Disposal Bin/DisposalBin.cs:30]: Worked?

I think you either mean "working" or "reworked"...

File: Assets/Content/Furniture/Machines/Kitchen/Microwave/Microwave.cs [hunk 6]
```diff
@@ -11,18 +11,30 @@
 using Mirror;
 using UnityEngine.Assertions;
 
+// Handles the microwave object
+// TODO: IMPORTANT, renamed the SND stuff to Sound
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Machines/Kitchen/Microwave/Microwave.cs:15]: "Renamed" should be "rename"

File: Assets/Content/Furniture/Machines/Vendors/Vendor.cs [hunk 7]
```diff
@@ -9,14 +9,16 @@
...
+    // Vendors are machines that can sell you stuff
     public class Vendor : InteractionTargetBehaviour
     {
         /// <summary>
         /// The point where items will be dispensed at
         /// </summary>
         public Transform EjectionPoint;
         /// <summary>
-        /// The velocity at which the item is launched
+        /// The velocity at which the item is launched, should be able to be hacked
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Machines/Vendors/Vendor.cs:21]: "Launched" --> "dispensed"

File: Assets/Content/Furniture/Machines/Vendors/Vendor.cs [hunk 8]
```diff
@@ -54,9 +63,12 @@ public override IInteraction[] GenerateInteractions(InteractionEvent interaction
             return new IInteraction[] {interaction};
         }
 
+	// Simple interaction to dispense items
+        // TODO: Once we have a menu for selection, this should throw the selected VendorItem
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Machines/Vendors/Vendor.cs:67]: "Throw" --> "dispense"

File: Assets/Content/Furniture/Machines/Vendors/Vendor.cs [hunk 9]
```diff
@@ -72,12 +84,16 @@ private void Dispense(InteractionEvent interactionEvent, InteractionReference re
...
+	    // TODO: Move this after the item has been spawned, for consistency reasons
             // Decrease item stock
             item.Stock--;
             
             // Create item
             GameObject gameObject = Instantiate(item.Prefab, EjectionPoint.position, EjectionPoint.rotation);
+	    // Spawn the item on the server
             NetworkServer.Spawn(gameObject);
+
+	    // Adds a little bit of force in the thrown item
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Machines/Vendors/Vendor.cs:96]: "In" --> "to"
"Thrown --> "dispensed"

File: Assets/Content/Furniture/Openable.cs [hunk 10]
```diff
@@ -6,18 +6,26 @@
 
 namespace SS3D.Content.Furniture
 {
+    // This handles a object or an item that can be opened
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Openable.cs:9]: "An object"

File: Assets/Content/Furniture/Openable.cs [hunk 11]
```diff
@@ -6,18 +6,26 @@
...
+    // by this requires the target to have a container to open
```
Comment (cosmiccoincidence) [inline Assets/Content/Furniture/Openable.cs:10]: Remove "by"

File: Assets/Content/Items/Functional/Tools/Generic/Crowbar.cs [hunk 12]
```diff
@@ -8,11 +8,16 @@
 
 namespace SS3D.Content.Items.Functional.Tools.Generic
 {
+    // This needs a good rework
     public class Crowbar : Item
     {
         public GameObject LoadingBarPrefab;
+
+	// nope
```
Comment (cosmiccoincidence) [inline Assets/Content/Items/Functional/Tools/Generic/Crowbar.cs:16]: ?

File: Assets/Engine/Inventory/Extensions/PickupInteraction.cs [hunk 13]
```diff
@@ -26,19 +29,25 @@ public Sprite GetIcon(InteractionEvent interactionEvent)
 
         public bool CanInteract(InteractionEvent interactionEvent)
         {
+            // if the target is whatever the hell Alain did
```
Comment (cosmiccoincidence) [inline Assets/Engine/Inventory/Extensions/PickupInteraction.cs:32]: @Alainx277

File: Assets/Engine/Inventory/Extensions/PickupInteraction.cs [hunk 14]
```diff
@@ -47,8 +56,11 @@ public bool CanInteract(InteractionEvent interactionEvent)
 
         public bool Start(InteractionEvent interactionEvent, InteractionReference reference)
         {
+	    // remember that when we call this Start, we are starting the interaction per see
```
Comment (cosmiccoincidence) [inline Assets/Engine/Inventory/Extensions/PickupInteraction.cs:59]: "per se"

File: Assets/Engine/Inventory/Item.cs [hunk 15]
```diff
@@ -104,6 +121,7 @@ public void GenerateNewIcon()
             sprite.name = transform.name;
         }
         
+	// this create the base interactions for an item, in this case, the drop interaction
```
Comment (cosmiccoincidence) [inline Assets/Engine/Inventory/Item.cs:124]: create --> creates

File: Assets/Engine/UI/MainMenuUI.cs [hunk 16]
```diff
@@ -12,15 +12,17 @@
 namespace SS3D.UI
 {
     /// <summary>
-    /// UI controller for join/host SS3D session
+    /// UI controller for join/host a SS3D session
```
Comment (cosmiccoincidence) [inline Assets/Engine/UI/MainMenuUI.cs:15]: for --> to

File: Assets/Content/Creatures/Creature.cs (hunk 1 above)
Comment (joaoburatto) [reply Assets/Content/Creatures/Creature.cs:11]: Correct, a ghost, while not visible or interactable, is controlled by the player and can interact with the map and entities (you can scare people and flick lights)

**Review by cosmiccoincidence — APPROVED** [review]
Approved. Not going to merge yet, will let others look at it if they want. You can merge in a day if no one else replies.

**Pull request merged**